ui/src/app/shared/vulnerability/vulnerabilities.component.ts

```diff
@@ -0,0 +1,48 @@
+import {Component, Input} from '@angular/core';
+import {Severity, Vulnerability} from '../../model/application.model';
+
+@Component({
+    selector: 'app-vulnerabilities',
+    templateUrl: './vulnerabilities.html',
+    styleUrls: ['./vulnerabilities.scss']
+})
+export class VulnerabilitiesComponent {
+
+    _vulnerabilities: Array<Vulnerability>;
+    summary = {
+        total: 0,
+        defcon: 0,
+        critical: 0,
+        high: 0,
+        medium: 0,
+        low: 0,
+        negligible: 0,
+        unknown: 0
+    };
+    @Input('vulnerabilities')
+    set vulnerabilities(data: Array<Vulnerability>) {
+        if (data) {
+            let defcon1 = data.filter(v => v.severity === Severity.DEFCON1 );
+            let critical = data.filter(v => v.severity === Severity.CRITICAL );
+            let high = data.filter(v => v.severity === Severity.HIGH );
+            let medium = data.filter(v => v.severity === Severity.MEDIUM );
+            let low = data.filter(v => v.severity === Severity.LOW );
+            let negligible = data.filter(v => v.severity === Severity.NEGLIGIBLE );
+            let unknown = data.filter(v => v.severity === Severity.UNKNOWN );
```

Comment:
I think it could be useful in this case to make a forEach loop in order to browse just one time the array. Because vulnerabilities could be very big. so a forEach with a push in each array is better